src/main/java/com/bookappstore/repository/order/OrderItemRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.bookappstore.repository.order;
+
+import com.bookappstore.model.OrderItem;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface OrderItemRepository extends JpaRepository<OrderItem, Long> {
+    @Query("SELECT oi FROM OrderItem oi JOIN oi.order o WHERE o.id = :orderId AND oi.id = :itemId")
```

Comment:
This query is redundant here. Please use the proper naming convention for method so that spring manages this operation for you. Fix the same issue in all places

Reply:
Thanks